### PR TITLE
refactor: Test enabling cleanups for code generator.

### PR DESCRIPTION
**Pull request by SandPod**
issue: https://github.com/serverpod/serverpod/issues/1151

Pure code refactor to make the code generator dependencies easier to understand and enables better testing.

Review commit by commit and pay attention to each refactor step.

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [x] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

None.

### CODE REVIEW

File: tools/serverpod_cli/lib/src/generator/dart/library_generators/entities_library_generator.dart [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import 'package:built_collection/built_collection.dart';
 import 'package:code_builder/code_builder.dart';
 import 'package:serverpod_cli/analyzer.dart';
-import '../shared.dart';
+import '../../shared.dart';
```
Comment (Isakdl) [inline tools/serverpod_cli/lib/src/generator/dart/library_generators/entities_library_generator.dart:4]: NIT: change to package import

**Review by Isakdl — APPROVED** [review]
LGTM :), two nit comments.

File: tools/serverpod_cli/lib/src/generator/dart/client_code_generator.dart [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import 'package:serverpod_cli/analyzer.dart';
+import 'package:serverpod_cli/src/generator/code_generator.dart';
+import 'package:path/path.dart' as p;
+import 'package:serverpod_cli/src/generator/dart/library_generators/entities_library_generator.dart';
+import 'package:serverpod_cli/src/generator/dart/library_generators/library_generator.dart';
+
+/// A [CodeGenerator] that generates the client side dart code of a
+/// serverpod project.
+class ClientCodeGenerator extends CodeGenerator {
```
Comment (vlidholt) [inline tools/serverpod_cli/lib/src/generator/dart/client_code_generator.dart:9]: DartClientCodeGenerator (may be other places too)

**Review by vlidholt — APPROVED** [review]
LGMT with comment fixed.

**Pull request merged**